streamlit_app/app.py

```diff
@@ -88,10 +90,23 @@ def fill_graph(nodes, edges, cypher_query):
 external_response = ""
 st.title("memary Demo")
 clear_memory = st.button("Clear Memory DB")
-query = st.text_input("Ask a question")
 
 
 
+#THIS will fail if the user does not have ollama downloaded - need to fix/find workaround
+models_info = ollama.list()
+available_models = tuple(model["name"] for model in models_info["models"])
+
+if "llama3:8b" in available_models:
+    selected_model = st.selectbox(
```

Comment:
Does this link back to self.llm?

Reply:
available models is grabbing from ollama installed models on current device
so no